pytorch_lightning/trainer/trainer.py

```diff
@@ -850,7 +850,7 @@ def call_setup_hook(self, model):
             called = self.datamodule.has_setup_test if self.testing else self.datamodule.has_setup_fit
             if not called:
                 self.datamodule.setup(stage_name)
-        self.setup(stage_name)
+        self.setup(model, stage_name)
```

Comment:
@Borda @tchaton  one thing I need to double check: if `setup` is called after DDP. do we need to unwrap `model` here? 

another fix would be be setting `Trainer.model` before `setup` is called. Which do you think is better?

Reply:
we're good here. setup gets called before wrapping the model in ddp